chat.c

```diff
@@ -255,6 +256,14 @@ proceed_chat_msg(char type, const char *author, const char *text) {
 		postuser = ">";
 	}
 
+	nicklen = strlen(nick);
+	if (type == 'c')
+		putchar('\a');
+	else if ((s = strstr(text, nick)) != NULL &&
+	    (s == text || !isalpha((unsigned char)s[-1])) &&
+	    !isalpha((unsigned char)s[nicklen]))
+		putchar('\a');
```

Comment:
This whole thing won't work correctly if stdout is not a terminal. While being unlikely, this still could happen, e.g., for logging purposes. I think that throwing a bell character in such case would do more harm, though. `isatty(STDOUT_FILENO)` check should do the trick.

Reply:
haven't thought about it, fixed.